### ISSUE TITLE
BUG: Tab order fails when expressions are in the fn position

### ISSUE THREAD

**Issue by schanzer** [bootstrapworld/codemirror-blocks#18]
To reproduce:
1) Open the blocks editor and add the expressions (+ 1 2) (- 3 4). All values are accessible via tab
2) Replace the addition function in the first expression with (- 3 4). Try tabbing through the values

EXPECTED
Tab order is outer-exp, inner-exp, -, 3, 4, 1, 2
ACTUAL
Tab order is outer-exp, inner-exp, 1, 2

**Comment by schanzer** [comment]
Looks like we're assuming only literals in the function position when building nodeMaps.

I believe the right fix is 827efc71bd8b4c21331d5ee60544215acd74b67f, but this generator stuff is new to me...
